src/components/seo/index.js

```diff
@@ -0,0 +1,86 @@
+import React from "react"
+import PropTypes from "prop-types";
+import {useLocation} from "@reach/router";
+import {useStaticQuery, graphql} from "gatsby";
+import {Helmet} from "react-helmet";
+
+export default function SEO({title, description, meta, lang}) {
+    const {pathname} = useLocation();
+
+    const {
+        defaultDescription,
+        defaultLanguage,
+        titleTemplate,
+        siteName,
+        siteUrl
+    } = siteMetadata();
+
+    const seo = {
+        url:         `${siteUrl}${pathname}`,
+        lang:        lang || defaultLanguage,
+        title:       title || siteName,
+        description: description || defaultDescription
+    }
+
+    return (
+        <Helmet 
+            htmlAttributes={{
+                lang: seo.lang
+            }}
+
+            title={seo.title}
+            titleTemplate={title ? titleTemplate : null}
+            meta={[
+                {
+                    name: "description",
+                    content: seo.description
+                },
+                {
+                    property: "og:site_name",
+                    content: siteName
+                },
+                {
+                    property: "og:title",
+                    content: title ? titleTemplate.replace(/%s/g, title) : seo.title
+                },
+                {
+                    property: "og:description",
+                    content: seo.description
+                }
+            ].concat(meta)}
+        />
+    )
+}
+
+SEO.propTypes = {
+    title: PropTypes.string,
+    description: PropTypes.string,
+    meta: PropTypes.arrayOf(PropTypes.object),
+    lang: PropTypes.string
+}
+
+SEO.defaultProps = {
```

Comment:
Default values can also be declared in the function signature.

For example:
```js
export default function SEO({title, description, meta = [], lang}) {
    ...
}
```
In this example the other values would be undefined if they are not present as a prop.
I've been using this pattern so far, so maybe we should stay consistent here, but I'm also open for debate on this.
I like the `SEO.propTypes` declaration, we should have that for all components.